### DISCUSSION TITLE
After updating Apollo to version 10.0.x of primereact - Some errors occur with the template.

### DISCUSSION THREAD

**Discussion by lordpizzo** [primefaces/community#159] [PrimeReact Templates] [answered]
After updating Apollo to version 10.0.x of primereact, the following errors started occurring in the AppConfig.tsx file:

./layout/AppConfig.tsx:57:21
Type error: Argument of type '(prevState: LayoutConfig) => { inputStyle: boolean; ripple: boolean; menuMode: MenuMode; menuTheme: MenuColorScheme; colorScheme: ColorScheme; theme: string; scale: number; }' is not assignable to parameter of type 'SetStateAction<LayoutConfig>'.
  Type '(prevState: LayoutConfig) => { inputStyle: boolean; ripple: boolean; menuMode: MenuMode; menuTheme: MenuColorScheme; colorScheme: ColorScheme; theme: string; scale: number; }' is not assignable to type '(prevState: LayoutConfig) => LayoutConfig'.
    Call signature return types '{ inputStyle: boolean; ripple: boolean; menuMode: MenuMode; menuTheme: MenuColorScheme; colorScheme: ColorScheme; theme: string; scale: number; }' and 'LayoutConfig' are incompatible.
      The types of 'inputStyle' are incompatible between these types.
        Type 'boolean' is not assignable to type 'string'.

```
 const changeInputStyle = (e: RadioButtonChangeEvent) => {
     setLayoutConfig((prevState) => ({ ...prevState, inputStyle: e.value }));

  };
```

  
  --------------------------------------  --------------------------------------  --------------------------------------
  The next Error:

Type error: Cannot invoke an object which is possibly 'undefined'.

    const changeRipple = (e: InputSwitchChangeEvent) => {
        setRipple(e.value !== undefined ? (e.value as boolean) : false)

     setLayoutConfig((prevState) => ({
          ...prevState,
          ripple: e.value as boolean
error Command failed with exit code 1.
  
 --------------------------------------  --------------------------------------  --------------------------------------

Type error: Argument of type '(prevState: LayoutConfig) => { menuMode: boolean; ripple: boolean; inputStyle: string; menuTheme: MenuColorScheme; colorScheme: ColorScheme; theme: string; scale: number; }' is not assignable to parameter of type 'SetStateAction<LayoutConfig>'.
  Type '(prevState: LayoutConfig) => { menuMode: boolean; ripple: boolean; inputStyle: string; menuTheme: MenuColorScheme; colorScheme: ColorScheme; theme: string; scale: number; }' is not assignable to type '(prevState: LayoutConfig) => LayoutConfig'.
    Call signature return types '{ menuMode: boolean; ripple: boolean; inputStyle: string; menuTheme: MenuColorScheme; colorScheme: ColorScheme; theme: string; scale: number; }' and 'LayoutConfig' are incompatible.
      The types of 'menuMode' are incompatible between these types.
        Type 'boolean' is not assignable to type 'MenuMode'.

```
  const changeMenuMode = (e: RadioButtonChangeEvent) => {
     setLayoutConfig((prevState) => ({ ...prevState, menuMode: e.value }))
                   
   }
```

**Comment by habubey** [comment]
We are aware of the issue and are working on a temporary fix until the new versions are released. I recommend downgrading your version until further notice. I will inform you as soon as possible.

**Comment by mertsincan** [comment]
@gucal Can you check this issue with the latest version and give an update to this ticket?

**Comment by gucal** [comment, marked as answer]
Hi,

These issues have been resolved with Apollo version 10.1.0.

Thank you for reporting.
